// @alipeople/sendon-sdk-typescript/dist/wrapper/wrapper.js
"use strict";
/* eslint-disable no-console */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Sendon = void 0;
const contacts_wrapper_1 = require("./contacts/contacts.wrapper");
const kakao_wrapper_1 = require("./kakao/kakao.wrapper");
const payment_wrapper_1 = require("./payment/payment.wrapper");
const point_wrapper_1 = require("./point/point.wrapper");
const rcs_wrapper_1 = require("./rcs/rcs.wrapper");
const sender_wrapper_1 = require("./sender/sender.wrapper");
const sms_wrapper_1 = require("./sms/sms.wrapper");
class Sendon {
    constructor(config) {
        this.debug = false;
        const configuration = {
            username: config.id,
            password: config.apikey,
            basePath: 'https://api.sendon.io',
        };
        this.debug = config.debug || false;
        this.sms = new sms_wrapper_1.SmsService(configuration, this.debug);
        this.kakao = new kakao_wrapper_1.KakaoService(configuration, this.debug);
        this.point = new point_wrapper_1.PointService(configuration, this.debug);
        this.sender = new sender_wrapper_1.SenderService(configuration, this.debug);
        this.contacts = new contacts_wrapper_1.ContactsService(configuration, this.debug);
        this.payment = new payment_wrapper_1.PaymentService(configuration, this.debug);
        this.rcs = new rcs_wrapper_1.RcsService(configuration, this.debug);
    }
}
exports.Sendon = Sendon;
__exportStar(require("../rest/api"), exports);
__exportStar(require("../rest/configuration"), exports);
__exportStar(require("./contacts/contacts.wrapper"), exports);
__exportStar(require("./kakao/kakao.wrapper"), exports);
__exportStar(require("./payment/payment.wrapper"), exports);
__exportStar(require("./point/point.wrapper"), exports);
__exportStar(require("./rcs/rcs.wrapper"), exports);
__exportStar(require("./sender/sender.wrapper"), exports);
__exportStar(require("./sms/sms.wrapper"), exports);
