// @alipeople/sendon-sdk-typescript/dist/wrapper/contacts/contacts.wrapper.js
"use strict";
/* eslint-disable no-console */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ContactsService = void 0;
const rest_1 = require("../../rest");
class ContactsService {
    constructor(configuration, debug) {
        this.contactsClient = new rest_1.SdoContacts(configuration);
        this.debug = debug;
    }
    async createBlocklist(params) {
        const response = await this.contactsClient.createBlocklist(params);
        if (this.debug)
            console.log(`response: ${JSON.stringify(response.data, null, 2)}`);
        return response.data;
    }
    async deleteBlocklist(blocklistId) {
        const response = await this.contactsClient.deleteBlocklist(blocklistId);
        if (this.debug)
            console.log(`response: ${JSON.stringify(response.data, null, 2)}`);
        return response.data;
    }
    async getBlocklist(params) {
        const response = await this.contactsClient.getBlocklist(params.senderNumber, params.kakaoChannelIds, params.blockType, params.startDate, params.endDate, params.limit, params.cursor);
        if (this.debug)
            console.log(`response: ${JSON.stringify(response.data, null, 2)}`);
        return response.data;
    }
}
exports.ContactsService = ContactsService;
