// @alipeople/sendon-sdk-typescript/dist/wrapper/sms/sms.wrapper.spec.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const api_1 = require("../../rest/api");
const config_1 = require("../config");
const sms_wrapper_1 = require("./sms.wrapper");
describe('sms', () => {
    let service;
    beforeAll(async () => {
        service = new sms_wrapper_1.SmsService({
            username: config_1.USERNAME,
            password: config_1.PASSWORD,
            basePath: config_1.BASEPATH,
        }, true);
    });
    it(`sms.send`, () => {
        const response = service.send({
            type: api_1.SmsMessageType.SMS,
            from: '01053601894',
            to: ['01053601894'],
            message: `센드온SDK를 이용한 문자 발송입니다.(${Math.random().toString(36).substring(2, 5)})`,
        });
        expect(response).toBeDefined();
    });
    it(`sms.sendExcel`, () => { });
    it(`sms.cancel`, () => { });
    it(`sms.find`, () => { });
    it(`sms.getMmsImageUrl`, () => { });
    it(`sms.uploadExcel`, () => { });
    it(`sms.uploadImages`, () => { });
});
