// @alipeople/sendon-sdk-typescript/dist/wrapper/sms/sms.wrapper.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SmsService = void 0;
/* eslint-disable no-console */
const rest_1 = require("../../rest");
class SmsService {
    constructor(configuration, debug) {
        this.smsClient = new rest_1.SdoSMS(configuration);
        this.debug = debug;
    }
    async send(params) {
        const response = await this.smsClient.send(params);
        if (this.debug)
            console.log(`response: ${JSON.stringify(response.data, null, 2)}`);
        return response.data;
    }
    async cancel(groupId) {
        const response = await this.smsClient.cancelGroup(groupId);
        if (this.debug)
            console.log(`response: ${JSON.stringify(response.data, null, 2)}`);
        return response.data;
    }
    async find(groupId) {
        const response = await this.smsClient.find(groupId);
        if (this.debug)
            console.log(`response: ${JSON.stringify(response.data, null, 2)}`);
        return response.data;
    }
    async getMmsImageUrl(params) {
        const response = await this.smsClient.getImageUrl(params.imageId1, params.expireIn, params.imageId3, params.imageId2);
        if (this.debug)
            console.log(`response: ${JSON.stringify(response.data, null, 2)}`);
        return response.data;
    }
    async uploadImages(params) {
        const response = await this.smsClient.uploadImages(params);
        if (this.debug)
            console.log(`response: ${JSON.stringify(response.data, null, 2)}`);
        return response.data;
    }
}
exports.SmsService = SmsService;
