// @alipeople/sendon-sdk-typescript/dist/esm/wrapper/sms/sms.wrapper.js
/* eslint-disable no-console */
import { SdoSMS, } from '../../rest';
export class SmsService {
    constructor(configuration, debug) {
        this.smsClient = new SdoSMS(configuration);
        this.debug = debug;
    }
    async send(params) {
        const response = await this.smsClient.send(params);
        if (this.debug)
            console.log(`response: ${JSON.stringify(response.data, null, 2)}`);
        return response.data;
    }
    async cancel(groupId) {
        const response = await this.smsClient.cancelGroup(groupId);
        if (this.debug)
            console.log(`response: ${JSON.stringify(response.data, null, 2)}`);
        return response.data;
    }
    async find(groupId) {
        const response = await this.smsClient.find(groupId);
        if (this.debug)
            console.log(`response: ${JSON.stringify(response.data, null, 2)}`);
        return response.data;
    }
    async getMmsImageUrl(params) {
        const response = await this.smsClient.getImageUrl(params.imageId1, params.expireIn, params.imageId3, params.imageId2);
        if (this.debug)
            console.log(`response: ${JSON.stringify(response.data, null, 2)}`);
        return response.data;
    }
    async uploadImages(params) {
        const response = await this.smsClient.uploadImages(params);
        if (this.debug)
            console.log(`response: ${JSON.stringify(response.data, null, 2)}`);
        return response.data;
    }
}
