// @alipeople/sendon-sdk-typescript/dist/esm/wrapper/wrapper.js
/* eslint-disable no-console */
import { ContactsService } from './contacts/contacts.wrapper';
import { KakaoService } from './kakao/kakao.wrapper';
import { PaymentService } from './payment/payment.wrapper';
import { PointService } from './point/point.wrapper';
import { RcsService } from './rcs/rcs.wrapper';
import { SenderService } from './sender/sender.wrapper';
import { SmsService } from './sms/sms.wrapper';
export class Sendon {
    constructor(config) {
        this.debug = false;
        const configuration = {
            username: config.id,
            password: config.apikey,
            basePath: 'https://api.sendon.io',
        };
        this.debug = config.debug || false;
        this.sms = new SmsService(configuration, this.debug);
        this.kakao = new KakaoService(configuration, this.debug);
        this.point = new PointService(configuration, this.debug);
        this.sender = new SenderService(configuration, this.debug);
        this.contacts = new ContactsService(configuration, this.debug);
        this.payment = new PaymentService(configuration, this.debug);
        this.rcs = new RcsService(configuration, this.debug);
    }
}
export * from '../rest/api';
export * from '../rest/configuration';
export * from './contacts/contacts.wrapper';
export * from './kakao/kakao.wrapper';
export * from './payment/payment.wrapper';
export * from './point/point.wrapper';
export * from './rcs/rcs.wrapper';
export * from './sender/sender.wrapper';
export * from './sms/sms.wrapper';
