// @alipeople/sendon-sdk-typescript/dist/wrapper/rcs/rcs.wrapper.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RcsService = void 0;
const api_1 = require("../../rest/api");
class RcsService {
    constructor(configuration, debug) {
        this.rcsClient = new api_1.SdoRCS(configuration);
        this.debug = debug;
    }
    async createCsvUploadUrl() {
        const response = await this.rcsClient.createCsvUploadUrl();
        if (this.debug)
            console.log(`response: ${JSON.stringify(response.data, null, 2)}`);
        return response.data;
    }
    async send(params) {
        const response = await this.rcsClient.send(params);
        if (this.debug)
            console.log(`response: ${JSON.stringify(response.data, null, 2)}`);
        return response.data;
    }
    async find(groupId) {
        const response = await this.rcsClient.find(groupId);
        if (this.debug)
            console.log(`response: ${JSON.stringify(response.data, null, 2)}`);
        return response.data;
    }
}
exports.RcsService = RcsService;
