// @alipeople/sendon-sdk-typescript/dist/wrapper/kakao/kakao.wrapper.spec.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const config_1 = require("../config");
const kakao_wrapper_1 = require("./kakao.wrapper");
describe('kakao', () => {
    let service;
    beforeAll(async () => {
        service = new kakao_wrapper_1.KakaoService({
            username: config_1.USERNAME,
            password: config_1.PASSWORD,
            basePath: config_1.BASEPATH,
        }, true);
    });
    // ================================
    // Message
    // ================================
    describe('sendAlimTalk', () => {
        it(`kakao.sendAlimTalk`, async () => {
            const response = await service.sendAlimTalk({
                sendProfileId: 'alipeople',
                templateId: '1234567890',
                to: [
                    {
                        phone: '01012345678',
                        variables: {
                            '#{고객명}': '신두온',
                        },
                    },
                ],
            });
            console.log(response);
            expect(response).toBeDefined();
        });
        it(`kakao.sendAlimTalk`, async () => {
            const response = await service.sendAlimTalk({
                sendProfileId: 'alipeople',
                templateId: '1234567890',
                to: [
                    {
                        phone: '01012345678',
                        variables: {
                            '#{고객명}': '신두온',
                        },
                    },
                ],
            });
            console.log(response);
            expect(response).toBeDefined();
        });
    });
    it(`kakao.sendFriendTalk`, () => { });
    // ================================
    // Template
    // ================================
    it(`kakao.delete`, () => { });
    it(`kakao.cancelReviewRequest`, () => { });
    it(`kakao.create`, () => { });
    it(`kakao.deleteAllByUserId`, () => { });
    it(`kakao.deleteTemplate`, () => { });
    it(`kakao.fetchCategories`, () => { });
    it(`kakao.findAll`, () => { });
    it(`kakao.findAllByUserId`, () => { });
    it(`kakao.findOneSendProfile`, () => { });
    it(`kakao.findOneTemplate`, () => { });
    it(`kakao.getProfileCategoryByCode`, () => { });
    it(`kakao.getTemplateCategories`, () => { });
    it(`kakao.getTemplateCategoryByCode`, () => { });
    it(`kakao.getTemplateSummary`, () => { });
    it(`kakao.registerProfile`, () => { });
    it(`kakao.requestReview`, () => { });
    it(`kakao.requestSendProfileTokenToKakao`, () => { });
    it(`kakao.updateDisplayName`, () => { });
    it(`kakao.updateTemplate`, () => { });
    it(`kakao.uploadCsv`, () => { });
    it(`kakao.uploadFriendTalkImage`, () => { });
    it(`kakao.uploadHighlightImage`, () => { });
    it(`kakao.uploadTemplateImage`, () => { });
    it(`kakao.uploadWideImage`, () => { });
});
