// @alipeople/sendon-sdk-typescript/dist/esm/wrapper/kakao/kakao.wrapper.js
/* eslint-disable no-console */
import * as rest from '../../rest';
// ===================================
// ===================================
// Kakako Wrapper
// ===================================
// ===================================
export class KakaoService {
    constructor(configuration, debug) {
        this.apiClient = new rest.SdoKakao(configuration);
        this.debug = debug;
    }
    // ===================================
    // Send Profile
    // ===================================
    async requestAuthToken(requestAuthTokenRequestSchema) {
        const response = await this.apiClient.requestAuthToken(requestAuthTokenRequestSchema);
        if (this.debug)
            console.log(`response: ${JSON.stringify(response.data, null, 2)}`);
        return response.data;
    }
    async registerChannel(registerChannelRequestSchema) {
        const response = await this.apiClient.registerChannel(registerChannelRequestSchema);
        if (this.debug)
            console.log(`response: ${JSON.stringify(response.data, null, 2)}`);
        return response.data;
    }
    async getSendProfiles() {
        const response = await this.apiClient.getSendProfiles();
        if (this.debug)
            console.log(`response: ${JSON.stringify(response.data, null, 2)}`);
        return response.data;
    }
    async getSendProfile(sendProfileId) {
        const response = await this.apiClient.getSendProfile(sendProfileId);
        if (this.debug)
            console.log(`response: ${JSON.stringify(response.data, null, 2)}`);
        return response.data;
    }
    // ===================================
    // Template
    // ===================================
    async createTemplate(sendProfileId, createTemplateRequestSchema) {
        const response = await this.apiClient.createTemplate(sendProfileId, createTemplateRequestSchema);
        if (this.debug)
            console.log(`response: ${JSON.stringify(response.data, null, 2)}`);
        return response.data;
    }
    async getTemplates(sendProfileId, query) {
        const response = await this.apiClient.getTemplates(sendProfileId, query === null || query === void 0 ? void 0 : query.keyword, query === null || query === void 0 ? void 0 : query.limit, query === null || query === void 0 ? void 0 : query.nextCursor, query === null || query === void 0 ? void 0 : query.status, query === null || query === void 0 ? void 0 : query.sort);
        if (this.debug)
            console.log(`response: ${JSON.stringify(response.data, null, 2)}`);
        return response.data;
    }
    async getTemplate(sendProfileId, templateId) {
        const response = await this.apiClient.getTemplate(sendProfileId, templateId);
        if (this.debug)
            console.log(`response: ${JSON.stringify(response.data, null, 2)}`);
        return response.data;
    }
    async updateTemplate(sendProfileId, templateId, updateTemplateRequestSchema) {
        const response = await this.apiClient.updateTemplate(sendProfileId, templateId, updateTemplateRequestSchema);
        if (this.debug)
            console.log(`response: ${JSON.stringify(response.data, null, 2)}`);
        return response.data;
    }
    async deleteTemplate(sendProfileId, templateId) {
        const response = await this.apiClient.deleteTemplate(sendProfileId, templateId);
        if (this.debug)
            console.log(`response: ${JSON.stringify(response.data, null, 2)}`);
        return response.data;
    }
    async requestReview(sendProfileId, templateId) {
        const response = await this.apiClient.requestReview(sendProfileId, templateId);
        if (this.debug)
            console.log(`response: ${JSON.stringify(response.data, null, 2)}`);
        return response.data;
    }
    async cancelReview(sendProfileId, templateId) {
        const response = await this.apiClient.cancelReview(sendProfileId, templateId);
        if (this.debug)
            console.log(`response: ${JSON.stringify(response.data, null, 2)}`);
        return response.data;
    }
    // ===================================
    // Message
    // ===================================
    async sendAlimTalk(sendAlimTalkRequestSchema) {
        const response = await this.apiClient.sendAlimTalk(sendAlimTalkRequestSchema);
        if (this.debug)
            console.log(`response: ${JSON.stringify(response.data, null, 2)}`);
        return response.data;
    }
    async sendBrandMessage(sendBrandMessageRequestSchema) {
        const response = await this.apiClient.sendBrandMessage(sendBrandMessageRequestSchema);
        if (this.debug)
            console.log(`response: ${JSON.stringify(response.data, null, 2)}`);
        return response.data;
    }
    async getGroup(groupId) {
        const response = await this.apiClient.getGroupInfo(groupId);
        if (this.debug)
            console.log(`response: ${JSON.stringify(response.data, null, 2)}`);
        return response.data;
    }
    async sendReservation(groupId) {
        const response = await this.apiClient.sendImmediate(groupId);
        if (this.debug)
            console.log(`response: ${JSON.stringify(response.data, null, 2)}`);
        return response.data;
    }
    async cancelReservation(groupId) {
        const response = await this.apiClient.cancelGroup(groupId);
        if (this.debug)
            console.log(`response: ${JSON.stringify(response.data, null, 2)}`);
        return response.data;
    }
    // ===================================
    // Upload Image
    // ===================================
    async uploadAlimTalkTemplateImage(image) {
        const response = await this.apiClient.uploadAlimTalkTemplateImage(image);
        if (this.debug)
            console.log(`response: ${JSON.stringify(response.data, null, 2)}`);
        return response.data;
    }
    async uploadFallbackImage(images) {
        const response = await this.apiClient.uploadFallbackImage(images);
        if (this.debug)
            console.log(`response: ${JSON.stringify(response.data, null, 2)}`);
        return response.data;
    }
    // ===================================
    // Brand Message Template Methods
    // ===================================
    /**
     * 브랜드메시지 텍스트형 템플릿 생성
     * @param sendProfileId 발신프로필 ID
     * @param request 템플릿 생성 요청 정보
     * @returns 생성된 템플릿 ID 및 코드
     */
    async createBrandMessageTextTemplate(sendProfileId, request) {
        const response = await this.apiClient.createTextBrandMessageTemplate(sendProfileId, request);
        if (this.debug)
            console.log(`response: ${JSON.stringify(response.data, null, 2)}`);
        return response.data;
    }
    /**
     * 브랜드메시지 이미지형 템플릿 생성
     * @param sendProfileId 발신프로필 ID
     * @param request 템플릿 생성 요청 정보
     * @returns 생성된 템플릿 ID 및 코드
     */
    async createBrandMessageImageTemplate(sendProfileId, request) {
        const response = await this.apiClient.createImageBrandMessageTemplate(sendProfileId, request);
        if (this.debug)
            console.log(`response: ${JSON.stringify(response.data, null, 2)}`);
        return response.data;
    }
    /**
     * 브랜드메시지 와이드형 템플릿 생성
     * @param sendProfileId 발신프로필 ID
     * @param request 템플릿 생성 요청 정보
     * @returns 생성된 템플릿 ID 및 코드
     */
    async createBrandMessageWideTemplate(sendProfileId, request) {
        const response = await this.apiClient.createWideBrandMessageTemplate(sendProfileId, request);
        if (this.debug)
            console.log(`response: ${JSON.stringify(response.data, null, 2)}`);
        return response.data;
    }
    /**
     * 브랜드메시지 와이드 아이템리스트형 템플릿 생성
     * @param sendProfileId 발신프로필 ID
     * @param request 템플릿 생성 요청 정보
     * @returns 생성된 템플릿 ID 및 코드
     */
    async createBrandMessageWideItemListTemplate(sendProfileId, request) {
        const response = await this.apiClient.createWideItemListBrandMessageTemplate(sendProfileId, request);
        if (this.debug)
            console.log(`response: ${JSON.stringify(response.data, null, 2)}`);
        return response.data;
    }
    /**
     * 브랜드메시지 캐러셀피드형 템플릿 생성
     * @param sendProfileId 발신프로필 ID
     * @param request 템플릿 생성 요청 정보
     * @returns 생성된 템플릿 ID 및 코드
     */
    async createBrandMessageCarouselFeedTemplate(sendProfileId, request) {
        const response = await this.apiClient.createCarouselFeedBrandMessageTemplate(sendProfileId, request);
        if (this.debug)
            console.log(`response: ${JSON.stringify(response.data, null, 2)}`);
        return response.data;
    }
    /**
     * 브랜드메시지 텍스트형 템플릿 수정
     * @param sendProfileId 발신프로필 ID
     * @param templateCode 템플릿 코드
     * @param request 템플릿 수정 요청 정보
     * @returns 수정된 템플릿 정보
     */
    async updateBrandMessageTextTemplate(sendProfileId, templateCode, request) {
        const response = await this.apiClient.updateTextBrandMessageTemplate(sendProfileId, templateCode, request);
        if (this.debug)
            console.log(`response: ${JSON.stringify(response.data, null, 2)}`);
        return response.data;
    }
    /**
     * 브랜드메시지 이미지형 템플릿 수정
     * @param sendProfileId 발신프로필 ID
     * @param templateCode 템플릿 코드
     * @param request 템플릿 수정 요청 정보
     * @returns 수정된 템플릿 정보
     */
    async updateBrandMessageImageTemplate(sendProfileId, templateCode, request) {
        const response = await this.apiClient.updateImageBrandMessageTemplate(sendProfileId, templateCode, request);
        if (this.debug)
            console.log(`response: ${JSON.stringify(response.data, null, 2)}`);
        return response.data;
    }
    /**
     * 브랜드메시지 와이드형 템플릿 수정
     * @param sendProfileId 발신프로필 ID
     * @param templateCode 템플릿 코드
     * @param request 템플릿 수정 요청 정보
     * @returns 수정된 템플릿 정보
     */
    async updateBrandMessageWideTemplate(sendProfileId, templateCode, request) {
        const response = await this.apiClient.updateWideBrandMessageTemplate(sendProfileId, templateCode, request);
        if (this.debug)
            console.log(`response: ${JSON.stringify(response.data, null, 2)}`);
        return response.data;
    }
    /**
     * 브랜드메시지 와이드 아이템리스트형 템플릿 수정
     * @param sendProfileId 발신프로필 ID
     * @param templateCode 템플릿 코드
     * @param request 템플릿 수정 요청 정보
     * @returns 수정된 템플릿 정보
     */
    async updateBrandMessageWideItemListTemplate(sendProfileId, templateCode, request) {
        const response = await this.apiClient.updateWideItemListBrandMessageTemplate(sendProfileId, templateCode, request);
        if (this.debug)
            console.log(`response: ${JSON.stringify(response.data, null, 2)}`);
        return response.data;
    }
    /**
     * 브랜드메시지 캐러셀피드형 템플릿 수정
     * @param sendProfileId 발신프로필 ID
     * @param templateCode 템플릿 코드
     * @param request 템플릿 수정 요청 정보
     * @returns 수정된 템플릿 정보
     */
    async updateBrandMessageCarouselFeedTemplate(sendProfileId, templateCode, request) {
        const response = await this.apiClient.updateCarouselFeedBrandMessageTemplate(sendProfileId, templateCode, request);
        if (this.debug)
            console.log(`response: ${JSON.stringify(response.data, null, 2)}`);
        return response.data;
    }
    /**
     * 브랜드메시지 템플릿 단건 조회
     * @param sendProfileId 발신프로필 ID
     * @param templateCode 템플릿 코드
     * @returns 템플릿 상세 정보
     */
    async getBrandMessageTemplate(sendProfileId, templateCode) {
        const response = await this.apiClient.getBrandMessageTemplate(sendProfileId, templateCode);
        if (this.debug)
            console.log(`response: ${JSON.stringify(response.data, null, 2)}`);
        return response.data;
    }
    /**
     * 브랜드메시지 템플릿 목록 조회
     * @param sendProfileId 발신프로필 ID
     * @param query 조회 조건 (선택)
     * @returns 템플릿 목록 및 총 개수
     */
    async getBrandMessageTemplates(sendProfileId, query) {
        const response = await this.apiClient.getBrandMessageTemplates(sendProfileId, query === null || query === void 0 ? void 0 : query.messageType, query === null || query === void 0 ? void 0 : query.status, query === null || query === void 0 ? void 0 : query.limit, query === null || query === void 0 ? void 0 : query.cursor);
        if (this.debug)
            console.log(`response: ${JSON.stringify(response.data, null, 2)}`);
        return response.data;
    }
    /**
     * 브랜드메시지 템플릿 삭제
     * @param sendProfileId 발신프로필 ID
     * @param request 삭제할 템플릿 ID 목록
     * @returns 삭제 결과
     */
    async deleteBrandMessageTemplate(sendProfileId, templateCode) {
        const response = await this.apiClient.deleteBrandMessageTemplate(sendProfileId, templateCode);
        if (this.debug)
            console.log(`response: ${JSON.stringify(response.data, null, 2)}`);
        return response.data;
    }
    /**
     * 브랜드메시지 이미지 업로드
     * @param file 업로드할 이미지 파일
     * @returns 업로드된 이미지 URL
     */
    async uploadBrandMessageImage(file) {
        const response = await this.apiClient.uploadBrandMessageImage(file);
        if (this.debug)
            console.log(`response: ${JSON.stringify(response.data, null, 2)}`);
        return response.data;
    }
    /**
     * 브랜드메시지 와이드 이미지 업로드
     * @param file 업로드할 와이드 이미지 파일
     * @returns 업로드된 이미지 URL
     */
    async uploadBrandMessageWideImage(file) {
        const response = await this.apiClient.uploadBrandMessageWideImage(file);
        if (this.debug)
            console.log(`response: ${JSON.stringify(response.data, null, 2)}`);
        return response.data;
    }
}
